### ISSUE TITLE
Document how to vendor *ring*

### ISSUE THREAD

**Issue by fabricedesre** [briansmith/ring#1058]
Steps to reproduce:
- create a project that uses Ring, verify that it builds.
- turn it into a vendored third party project, eg. running `cargo vendor third-party/`
- build your project.

Expected:
- Build success

Observed:
- Build failure such as:
```
  thread 'main' panicked at '"/home/fabrice/dev/ring-vendored/third-party/ring/pregenerated/aesni-x86_64-elf.S": Os { code: 2, kind: NotFound, message: "No such file or directory" }', /home/fabrice/dev/ring-vendored/third-party/ring/build.rs:778:10
```

**Comment by briansmith** [comment]
Does `cargo vendor` vendor the entire contents of the Cargo package? When I publish *ring* to crates.io, I run a pre-processing step that generates the pregenerated asm files. Are those pregenerated files being included into third-party by `cargo vendor`?

**Comment by fabricedesre** [comment]
No, the content of the `pregenerated/` directory is not included by `cargo vendor`.

**Comment by briansmith** [comment]
> No, the content of the pregenerated/ directory is not included by cargo vendor.

I think other people have got it to work. https://github.com/alexcrichton/cargo-vendor/issues/89 shows that it doesn't work so well who tried to vendor a fork from git, but if you vendor the crate from crates.io, does it work?

**Comment by fabricedesre** [comment]
Yes I think it works from crates.io, but unfortunately I'm on a fork. Anyway I have a local workaround, and I can understand that you don't want changes for the case of git forks. Feel free to close!

**Comment by briansmith** [comment]
> Yes I think it works from crates.io, but unfortunately I'm on a fork. Anyway I have a local workaround, and I can understand that you don't want changes for the case of git forks. Feel free to close!

I will document how to "properly" vendor *ring* such that, if you follow those directions then it should work.

**Comment by briansmith** [comment]
In the interim:

This is *the* one way I fully support vendoring *ring*: Use `cargo vendor` to vendor the latest stable version of *ring* published from crates.io.

If you aren't going to use the one fully supported way, then, on a Windows machine with the necessary tools installed, run mk/package.sh and vendor the resulting package with the pregenerated code.

**Comment by briansmith** [comment]
Another way that seems to work is to use `cargo clone`, https://github.com/JanLikar/cargo-clone.

```
$ cargo clone $CRATE --vers $VERSION --prefix vendored
```

Basically, use a command that gets the source from crates.io, instead of from Git.
